File: 2-Open-Closed-Principle (OCP)/python/OCP-bad-example.py
# Bad Example
class Post:
    def __init__(self, content):
        self.content = content

    def publish(self):
        # Logic to publish the post
        pass

    # Later, adding a new type of post directly in Post class
    def publish_image_post(self, image_url):
        # Logic specific to publishing an image post
        pass

File: 2-Open-Closed-Principle (OCP)/js/OCP-bad-example.js
// Bad Example
class Post {
  constructor(content) {
    this.content = content;
  }

  publish() {
    // Logic to publish the post
  }
}

// Later, adding a new type of post directly in Post class
Post.prototype.publishImagePost = function(imageUrl) {
  // Logic specific to publishing an image post
};